WBooks/BookRepository.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  BookRepository.swift
+//  WBooks
+//
+//  Created by Diego Quirós on 3/14/18.
+//  Copyright © 2018 Wolox. All rights reserved.
+//
+
+import Foundation
+import Networking
+import Argo
+import Result
+import ReactiveSwift
+import GoogleSignIn
+
+protocol BookRepositoryType {
+    
+    func fetchBooks() -> SignalProducer<[Book], RepositoryError>
+    
+}
+
+internal class BookRepository: AbstractRepository, BookRepositoryType {
+    
+    let token = GoogleLoginService.shared.sessionManager.sessionToken
```

Comment:
The coupling with the `GoogleLoginService` is not necessary and should be avoided. As a good practice I would suggest to only depend on singletons when providing default parameters in the class / struct initializer. The `BookRepository` already receives a session manager via its initializer. This line could be changed to

```swift
var token: String {
    return sessionManager.token
}
```